src/main/java/org/mastodon/model/branch/BranchGraphHighlightAdapter.java

```diff
@@ -48,28 +52,54 @@ public class BranchGraphHighlightAdapter<
 
 	private final HighlightModel< V, E > highlight;
 
+	private final TimepointModel timepoint;
+
 	public BranchGraphHighlightAdapter(
 			final BranchGraph< BV, BE, V, E > branchGraph,
 			final ReadOnlyGraph< V, E > graph,
 			final GraphIdBimap< V, E > idmap,
-			final HighlightModel< V, E > highlight )
+			final HighlightModel< V, E > highlight,
+			final TimepointModel timepoint )
 	{
 		super( branchGraph, graph, idmap );
 		this.highlight = highlight;
+		this.timepoint = timepoint;
 	}
 
 	@Override
-	public void highlightVertex( final BV vertex )
+	public void highlightVertex( final BV branchVertex )
 	{
-		if ( null == vertex )
+		if ( null == branchVertex )
 		{
 			highlight.highlightVertex( null );
 			return;
 		}
 
-		final V vRef = graph.vertexRef();
-		highlight.highlightVertex( branchGraph.getLastLinkedVertex( vertex, vRef ) );
-		graph.releaseRef( vRef );
+		final V spotRef = graph.vertexRef();
+		Iterator< V > vertices = branchGraph.vertexBranchIterator( branchVertex );
+		boolean highlighted = false;
+		try
+		{
+			// prefer to highlight the spot at the current timepoint
+			while ( vertices.hasNext() )
```

Comment:
This might have too high of a cost on performance. I am afraid that it would compromise the responsiveness.

Reply:
> Hm, I admit that for the showing the last spot (as in the current implementation), there is a direct shortcut with a map holding the first/last spot for each branchspot.

Ah? Could you clarify this point?

>On the other side: typical tracking projects rarely have branchspots longer than 1000 spots. And for every highlighting, it would only be required to iterate through the spots of only the one branchspot, which is highlighted. Thus, normally it would not be too much of iterations required.

Yes you are right. Let's make it work.